doc/modules/preprocessing.rst

```diff
@@ -474,11 +474,11 @@ then the shape of the transformed input is
     Imputer(add_indicator_features=True, axis=0, copy=True, missing_values='NaN',
         strategy='mean', verbose=0)
     >>> print(imp_with_in.transform(X))                           # doctest: +ELLIPSIS
-    [[ 4.          2.          1.          0.        ]
-     [ 6.          3.66666667  0.          1.        ]
+    [[ 4.          2.          0.          0.        ]
```

Comment:
I'd like you to have a go, but showing what you've got early - and writing tests to show your understanding - helps make explicit checkpoints.

I was thinking a couple of hours ago about the case where a feature has missing values at `transform` time but not at `fit`. I do think this is a possible case, but I also think it's silly to add an indicator column for each of thousands of features not requiring imputation. We could actually allow the user to specify which features may require imputation (and hence both statistics and indicator matrix), with a default to infer them from the training data. What do you think?